Create Customer class and Runner class, both of which extends Person …

melvintzw opened the pull request:
…class.

Addresses issue #57

melvintzw committed:
Create Customer class and Runner class, both of which extends Person class.

jonleeyz commented on src/main/java/seedu/address/model/person/Customer.java:
```diff
@@ -0,0 +1,100 @@
+    private final Address address;
+
+    private final UniqueTagList tags;
+
```
Would declaration of these attributes be redundant, since they are already inherited from the `Person` parent class?

jonleeyz commented on src/main/java/seedu/address/model/person/Runner.java:
```diff
@@ -0,0 +1,85 @@
+    private final Address address;
+
+    private final UniqueTagList tags;
+
```
Would declaration of these attributes be redundant, since they are already inherited from the `Person` parent class?

melvintzw replied:
oh haha I just forgot how construct existing fields by referencing the superclass, I'm checking out the java documentaion and tutorials now and I think I know how to do it now, thanks!

jonleeyz commented on src/main/java/seedu/address/model/person/Customer.java:
```diff
@@ -0,0 +1,100 @@
+        this.tags = new UniqueTagList(tags);
+        this.moneyOwed = moneyOwed;
+        this.interestRate = interestRate;
+    }
```
Could we can call the `Person` constructor to do the assignment of attributes inherited from the `Person` parent class?

melvintzw replied:
we could but I looked through the rest of the code and it might make more sense to/be easier to construct a Customer directly rather than construct a Person than a Customer. I'm not sure though, need to discuss this further.

jonleeyz commented on src/main/java/seedu/address/model/person/Person.java:
```diff
@@ -25,6 +25,16 @@
+        this.email = new Email("default@example.com");
+        this.address = new Address("default address");
+        this.tags = new UniqueTagList();
+
```
Extra whitespace: redundant?

jonleeyz commented on src/main/java/seedu/address/model/person/Runner.java:
```diff
@@ -0,0 +1,85 @@
+        this.address = address;
+        this.tags = new UniqueTagList(tags);
+
+    }
```
Could we can call the `Person` constructor to do the assignment of attributes inherited from the `Person` parent class?

jonleeyz commented on src/main/java/seedu/address/model/person/Runner.java:
```diff
@@ -0,0 +1,85 @@
+        this.email = email;
+        this.address = address;
+        this.tags = new UniqueTagList(tags);
+
```
Extra whitespace: redundant?

jonleeyz commented on src/main/java/seedu/address/model/person/Runner.java:
```diff
@@ -0,0 +1,85 @@
+    /**
+     * Returns an immutable tag set, which throws {@code UnsupportedOperationException}
+     * if modification is attempted.
+     */
```
Is this comment block supposed to be somewhere else? Seems "lost" haha.

jonleeyz commented on src/main/java/seedu/address/model/person/Person.java:
```diff
@@ -25,6 +25,16 @@
     /**
      * Every field must be present and not null.
      */
+
```
Would it be possible for us to make the `Person` class abstract?

melvintzw replied:
Possible, just wondering if we want to keep a Person class!

jonleeyz reviewed (dismissed):
I think there could be some changes we could make to fully utilise inheritance and prevent code duplication.

But otherwise, great job @melvintzw :)

jonleeyz committed:
Merge branch 'master' into LoansharkModel

melvintzw committed:
change fields

melvintzw committed:
Add one line of pseudocode

melvintzw committed:
Merge branch 'LoansharkModel' of https://github.com/melvintzw/main into LoansharkModel

melvintzw committed:
add natty API
add various fields for Customer
add method for calculating money owed after weekly compounded interest

melvintzw committed:
change formula for Customer.getMoneyCurrentlyOwed()
add JUnit test to test correctness of getMoneyCurrentlyOwed method

melvintzw committed:
Merge branch 'master' into LoansharkModel

melvintzw committed:
rename test method

melvintzw committed:
create Runner class

melvintzw committed:
modify toString() in Customer and Runner

melvintzw committed:
Merge branch 'LoansharkModel' of https://github.com/melvintzw/main into LoansharkModel

melvintzw committed:
Remove unused import

melvintzw committed:
change erroneous default constructor in Person

melvintzw committed:
Encapsulate Customer fields
Create customer and runner package within person package for better readability and organisation of files
Create default constructors for Phone, Email, Address, with their values as empty String
Modify CustomerTest to work with new changes

melvintzw committed:
Fix checkstyle error: wrong lexographical order of import

zhangriqi reviewed (approved):
Great
